livox_ros_driver/livox_ros_driver/include/livox_ros_driver_nodelet.h

```diff
@@ -0,0 +1,60 @@
+//
+// The MIT License (MIT)
+//
+// Copyright (c) 2019 Livox. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+
+#ifndef LIVOX_ROS_DRIVER_INClUDE_LIVOX_ROS_DRIVER_NODELET_H_
+#define LIVOX_ROS_DRIVER_INClUDE_LIVOX_ROS_DRIVER_NODELET_H_
+
+#include "livox_ros_driver.h"
+#include "../lddc.h"
+
+#include <ros/ros.h>
+#include <nodelet/nodelet.h>
+#include <thread>
+#include <memory>
+
+namespace livox_ros
+{
+
+class LivoxRosDriverNodelet : public nodelet::Nodelet
+{
+  public:
+    LivoxRosDriverNodelet();
+
+    ~LivoxRosDriverNodelet();

```

Comment:
Same here, since the std::thread destructor correctly handles the joining of threads. 
```
~LivoxRosDriverNodelet() = default;
```

Reply:
Do you think we should remove the definition of the destructor? If we don't do it we'd get:

```
/home/borghi/ws_core/src/livox_ros_driver/livox_ros_driver/livox_ros_driver/livox_ros_driver_nodelet.cpp:47:3: error: definition of explicitly-defaulted ‘livox_ros::LivoxRosDriverNodelet::~LivoxRosDriverNodelet()’
   LivoxRosDriverNodelet::~LivoxRosDriverNodelet()
   ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/borghi/ws_core/src/livox_ros_driver/livox_ros_driver/livox_ros_driver/livox_ros_driver_nodelet.cpp:25:0:
/home/borghi/ws_core/src/livox_ros_driver/livox_ros_driver/livox_ros_driver/include/livox_ros_driver_nodelet.h:44:5: note: ‘virtual livox_ros::LivoxRosDriverNodelet::~LivoxRosDriverNodelet()’ explicitly defaulted here
     ~LivoxRosDriverNodelet() = default;
```

Reply:
so by defaulting the constructor/deleter, we no longer need the implementations in the C++ file.

Reply:
I understand the purpose of the `default` keyword but what I want to know if it's you want me to also remove the implementation that it's defined in `livox_ros_driver_nodelet.cpp`:

```
LivoxRosDriverNodelet::~LivoxRosDriverNodelet()
  {
    if (thread_->joinable())
    {
      thread_->join();
    }
  }
```

Reply:
After a bit more reading, it seems that what I'm suggesting isn't the safest option. 
https://stackoverflow.com/questions/13983984/what-happens-when-calling-the-destructor-of-a-thread-object-that-has-a-condition

so for now, let's just leave the destructor implementation and not default it. The constructor can still be defaulted.